File: leetcode/383.赎金信/383.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int cnt[26] = { 0 };
        for (int i = 0; i < magazine.size(); i++) {
            cnt[magazine[i] - 'a']++;
        }
        for (int i = 0; i < ransomNote.size(); i++) {
            cnt[ransomNote[i] - 'a']--;
        }
        for (const int& c : cnt) {
            if (c < 0)
                return false;
        }
        return true;
    }
};

File: leetcode/383.赎金信/383.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        cnt = [0]*26
        for i in magazine:
            cnt[ord(i) -ord("a")] += 1
        for i in ransomNote:
            cnt[ord(i)- ord("a")] -= 1
        for i in cnt:
            if i < 0:
                return False
        return True